CRM/Contact/Form/Search.php

```diff
@@ -913,13 +913,21 @@ protected function loadMetadata() {
     if (self::checkComponentAccess('CiviContribute')) {
       $this->addSearchFieldMetadata(['Contribution' => CRM_Contribute_BAO_Query::getSearchFieldMetadata()]);
       $this->addSearchFieldMetadata(['ContributionRecur' => CRM_Contribute_BAO_ContributionRecur::getContributionRecurSearchFieldMetadata()]);
+      $this->addSearchFieldMetadata(['Pledge' => CRM_Pledge_BAO_Query::getSearchFieldMetadata()]);
```

Comment:
Access CiviPledge is a separate permission. Otherwise the PR looks good

Reply:
ok fixed @eileenmcnaughton